jetty-util/src/main/java/org/eclipse/jetty/util/TypeUtil.java

```diff
@@ -751,4 +756,52 @@ public T next()
             }
         };
     }
+
+    /**
+     * Used on a {@link ServiceLoader#stream()} with {@link Stream#flatMap(Function)},
+     * so that in the case a {@link ServiceConfigurationError} is thrown it warns and
+     * continues iterating through the service loader.
+     * <br>Usage Example:
+     * <p>{@code ServiceLoader.load(Service.class).stream().flatMap(TypeUtil::providerMap).collect(Collectors.toList());}</p>
+     * @param <T> The class of the service type.
+     * @param provider The service provider to instantiate.
+     * @return a stream of the loaded service providers.
+     */
+    public static <T> Stream<T> providerMap(ServiceLoader.Provider<T> provider)
```

Comment:
The plan was to have `mapToService` as public because when the JVM implementation for `ServiceLoader.stream()` is fixed we could change our code to instead use `serviceLoader.stream().flatMap(TypeUtil::mapToService);` and remove our implementations for `serviceStream(ServiceLoader<T>)` and `serviceProviderStream(ServiceLoader<T>)`.